### PR TITLE
feat(hybrid-cloud): scim endpoints -> hc test bed

### PR DESCRIPTION
**Pull request by corps**
https://getsentry.atlassian.net/browse/HC-204

Goal of the https://www.notion.so/sentry/Infra-Automation-Hybrid-Cloud-68b647b96575463ca5b3e6602da6c470 test bed milestone is to have a provisioned deployment of the hybrid cloud region / control silo split in production, and *have it be accepting real, meaningful data, and be used / dogfooded*. 

Tentative plan here is instrument SCIM endpoints which are owned by the enterprise team to send exceptions to a configurable, new DSN that will resolve to this potential provisioned endpoint.

This is of course "off by default" and still depends on having such a test bed dsn configured.  However, this change should otherwise be entirely safe no op on merge.

I followed the performance team's example in using the multiplexing transport for dynamic dsn swapping.  Let me know if there is any more preferable way to do this.

### CODE REVIEW

File: src/sentry/options/defaults.py [hunk 1]
```diff
@@ -538,3 +538,7 @@
 # region to control silo multi region mode controls
 # It is safe to switch this off or on freely, it is purely for testing features ahead of hybrid cloud rollout.
 register("hc.region-to-control.monolith-publish", default=False)
+
+# Killswitch for sending tagged internal errors to the hybrid cloud test bed dsn
+# Note: A value that is neither 0 nor 1 is regarded as 0
+register("store.use-hc-test-bed-dsn-sample-rate", default=0)
```
Comment (k-fish) [inline src/sentry/options/defaults.py:544]: I feel like it's not really being used as a 'rate' here and more of a bool to act as a killswitch 🤔, also we usually create options first as a separate PR, then have it initialized so it's cached on deploy for when it's accessed.

**Review by k-fish — COMMENTED** [review]
Alternatively, since the performance team no longer needs to use the experiment dsn, you can re-use the experimental transport code and save having to make a new option etc. You'd just need to switch the dsn it currently points out and set it's option back to 0.0 before you merge.

File: src/sentry/utils/sdk.py [hunk 2]
```diff
@@ -120,6 +120,7 @@
 
 UNSAFE_TAG = "_unsafe"
 EXPERIMENT_TAG = "_experimental_event"
+HYBRID_CLOUD_TEST_BED_TARGET_TAG = "_hybrid_cloud_test_bed"
```
Comment (markstory) [inline src/sentry/utils/sdk.py:123]: I don't understand why we are `_` prefixing tag values for our own application. Its not like we need to worry about conflicts with other parts of our application.

Comment (corps) [reply src/sentry/utils/sdk.py:123]: I also don't understand, but I used the rule of "follow patterns established" when in doubt 🤷 

Possibly, the idea was also to have these tags sort differently so as not to mix it with other categorically different tags???

Comment (markstory) [reply src/sentry/utils/sdk.py:123]: Could be a sorting thing. I'm not sure why any of these tags have `_` prefixes. It isn't that important though.

File: src/sentry/options/defaults.py (hunk 1 above)
Comment (rgibert) [reply src/sentry/options/defaults.py:544]: New options will get cached on first retrieval (see https://github.com/getsentry/sentry/pull/39799).  Not sure if that changes us wanting to split out the new option.

**Pull request closed without merging**